_posts/2016-4-2-code-splitting-angular2.md

```diff
@@ -6,61 +6,97 @@ date:   2016-4-2
+Webpack 2 provides us with `System.import('../path/to/file.js')` syntax (and soon [`import` syntax](https://github.com/tc39/proposal-dynamic-import)) to achieve loading of the additional chunks via a Promise based api. Webpack processes `System.import` by compiling `file.js` and its dependencies into a separate file and then dynamically loading that code when that promise is resolved.
 
 ```js
-  const importedChunk = System.import('./foo').then((foo) => foo);
+  const importedChunk = System.import('./foo').then((foo) => foo.default);
 ```
 
-Now we have a way to split and load the pieces of our application, but we still need to know where to spilt. It ultimately depends on your app, but it's generally a good idea to split up your app based on views or pages and load the chunks dynamically on navigation. With single page apps, you can think of a main page and profile page and loading the code for the profile page when you navigate from the main page.
+We have a way to split and load the pieces of our application, but we still need to know where to split. It ultimately depends on your app, but it's generally a good idea to split up your app based on views or pages and load the additional chunks dynamically on navigation. With single page apps, you can think of a main page and profile page and loading the code for the profile page when you navigate from the main page.
 
 Let's say we have the following example Angular 2 application.
 
-#### App.ts
+#### AppModule.ts
 
 ```js
-// App.ts
-import { Component } from 'angular2/core';
-import { RouteConfig, ROUTER_DIRECTIVES } from 'angular2/router';
-import InlineSubComp from './InlineSubComp';
+// AppModule.ts
+import { NgModule } from '@angular/core';
+import { BrowserModule } from '@angular/platform-browser';
+
+import { routes } from './Routing';
+
+import AppComponent from './AppComponent';
+import InlineComp from './InlineComp';
+
+@NgModule({
+  declarations: [
+    AppComponent,
+    InlineComp
+  ],
+  imports: [ BrowserModule, routes ],
+  bootstrap: [ AppComponent ]
+})
+export default class AppModule {}
+
+```
+
+#### AppComponent.ts
+
+```js
+// AppComponent.ts
+import { Component } from '@angular/core';
 
 @Component({
   selector: 'yo',
   template: `
-  <div>
-    <h1>Code Splitting in Angular 2</h1>
+    <h1>yo</h1>
     <router-outlet></router-outlet>
-  </div>
-  `,
-  directives: [ ROUTER_DIRECTIVES ]
+  `
 })
-@RouteConfig([
-  { path: '/', name: 'Root', component: InlineSubComp }
-])
-export default class App {}
+export default class AppComponent {}
 ```
 
 #### InlineComp.ts
 
 ```js
 // InlineSubComp.ts
-import { Component } from 'angular2/core';
+import { Component } from '@angular/core';
 
 @Component({
-  selector: 'inline-sub-comp',
-  template: `<div>This component was loaded inline.</div>`
+  selector: 'inline-comp',
```

Comment:
it appears that omitting the selector just puts a generic `ng-module` tag instead of `inline-comp`. I feel like `inline-comp` is more specific and identifies DOM associated to this component. Keeping as is.

<img width="633" alt="screen shot 2017-01-05 at 11 32 33 am" src="https://cloud.githubusercontent.com/assets/434885/21692567/b0295814-d33a-11e6-975c-2e94fffd84f9.png">